Submitted By: Ryan Oliver <ryan.oliver@pha.com.au>
Date: 2005-02-04
Initial Package Version: 3.2.5
Upstream Status:
Origin: Jim Gifford, Ryan Oliver
Description: Fixes procps build for mips32 with glibc > 20041124
(changes in sysdeps/mips/sgidefs.h). Don't only test whether _ABIN32
is defined, also test that it matches _MIPS_SIM before setting KLONG.

diff -uNr procps-3.2.5-orig/proc/procps.h procps-3.2.5/proc/procps.h
--- procps-3.2.5-orig/proc/procps.h	2004-08-20 15:40:13 -0700
+++ procps-3.2.5/proc/procps.h	2005-02-02 22:46:11 -0800
@@ -31,7 +31,7 @@
 //
 // Unknown: PA-RISC and zSeries
 //
-#if defined(k64test) || defined(_ABIN32)
+#if defined(k64test) || (defined(_ABIN32) && _MIPS_SIM == _ABIN32)
 #define KLONG long long    // not typedef; want "unsigned KLONG" to work
 #define KLF "L"
 #define STRTOUKL strtoull
